lib/Client.js

```diff
@@ -1042,31 +1054,104 @@ class Client extends EventEmitter {
         }).then((instance) => new StageInstance(instance, this));
     }
 
+    /**
+     * Create a thread in a channel
+     * @arg {String} channelID The ID of the channel
+     * @arg {Object} [options] The thread options
```

Comment:
```suggestion
     * @arg {Object} options The thread options
```
This isn't optional, `options.name` is a required property (and also your code doesn't allow for it to be optional, but besides the point)